web/web.go

```diff
@@ -0,0 +1,42 @@
+package web
+
+import (
+	"net/http"
+	"io/ioutil"
+	"html/template"
+)
+
+func init() {
+	http.HandleFunc("/", rootHandler)
+	http.HandleFunc("/results", resultsHandler)
+}
+
+func rootHandler(writer http.ResponseWriter, request *http.Request) {
+	parseTemplate("search", writer, request)
+}
+
+func resultsHandler(writer http.ResponseWriter, request *http.Request) {
+	parseTemplate("results", writer, request)
+}
+
+func parseTemplate(templateName string, writer http.ResponseWriter, request *http.Request) {
+	htmlTemplate, err := loadTemplate(templateName, writer)
```

Comment:
One thing you might want to consider is either passing in the template name to the loadTemplate function, or using the templateName as the name you pass to template.New. This just removes a magic string